src/SourceLocator/LocatedSource.php

```diff
@@ -31,7 +33,7 @@ public function __construct($source, $filename)
             );
         }
 
-        if (null !== $filename) {
+        if (null !== $filename && self::INTERNAL_SOURCE_MAGIC_CONST !== $filename) {
```

Comment:
yeah, I wasn't sure how to do it, but it's basically to pass the flag down to `ReflectionClass->isInternal()` method so it knows internal or not